playbooks/duplicate-hypervisor.yml

```diff
@@ -0,0 +1,30 @@
+---
+# Copyright 2018, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Check duplicate hypervisor
+  hosts: utility_all[0]
+  tasks:
+    - name: List nova hypervisor
+      shell: |
+        . {{ ansible_env.HOME }}/openrc
+        nova {{ openrc_insecure |default(false) |bool |ternary('--insecure','') }} hypervisor-list
+      args:
+        executable: /bin/bash
+      register: nova_hypervisorlist_result
+
+    - name: Verify duplicate hypervisor
+      fail:
+        msg: "Mulitple hypervisor existing!"
+      when: "{{ nova_hypervisorlist_result.stdout_lines | length == 5 }}"
```

Comment:
I don't understand why length is hard coded to 5 and we are not sorting/counting by service name for find duplicate hosts ?

Reply:
Since a list with single hypervisor only has 5 lines in stdout.lines. I know it's kinda dumm but could you give more intelligent way to do this?